algorithms/src/std_algorithms/Kokkos_MinMaxElementOperations.hpp

```diff
@@ -0,0 +1,438 @@
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef KOKKOS_STD_MIN_MAX_ELEMENT_OPERATIONS_HPP
+#define KOKKOS_STD_MIN_MAX_ELEMENT_OPERATIONS_HPP
+
+#include <Kokkos_Core.hpp>
+#include "Kokkos_BeginEnd.hpp"
+#include "Kokkos_StdAlgorithmsConstraints.hpp"
+#include "Kokkos_ModifyingSequenceOperations.hpp"
+
+/// \file Kokkos_MinMaxElementOperations.hpp
+/// \brief Kokkos min/max element operations
+
+namespace Kokkos {
+namespace Experimental {
+// see https://github.com/kokkos/kokkos/issues/4075
+
+// ------------------------------------------
+// begin Impl namespace
+// ------------------------------------------
+namespace Impl {
+
+template <class IteratorType>
+struct _StdAlgoMinMaxOpsDefaultLessThanComparator {
+  using value_type = typename IteratorType::value_type;
+  KOKKOS_INLINE_FUNCTION
+  bool operator()(const value_type& a, const value_type& b) const {
+    return a < b;
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  bool operator()(const volatile value_type& a,
+                  const volatile value_type& b) const {
+    return a < b;
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  _StdAlgoMinMaxOpsDefaultLessThanComparator() {}
```

Comment:
Because std algo need a custom comparator, so I have to implement a few custom reducers since those already present were hard-wired to use `operator<` which is too restrictive.

Reply:
default constructor removed